nxt/sensor/hitechnic.py

```diff
@@ -459,9 +460,238 @@ def set_digital_modes(self, modes):
+        """
+        Get digital input pins (D0-D7)
+        :return: Boolean list, LSB first.
+        """
+        digital_in = self.read_value("digital_in")[0]
+        boolean_list = self._byte_to_boolean_list(digital_in, reverse=True)
+        digital_in_map = {}
+        for x in range(0, 8):
+            digital_in_map["b{}".format(x)] = boolean_list[x]
+        return digital_in_map
+
+    def set_digital(self, pins: list[bool]):
+        """
+        Set digital output pins (D0-D7)
+        :param pins: boolean list (LSB first)
+        """
+        if len(pins) != 8:
+            raise RuntimeError("Need to specify list of 8 boolean values")
+        self.write_value("digital_out", [self._boolean_list_to_byte(pins)])
+
+    def set_digital_byte(self, integer: int, lsb=True):
+        """
+        Set digital output pins (D0-D7) from byte
+        :param integer: Byte (0-255 inclusive)
+        :param lsb: Whether to output in LSB order (0x01 = pin B0)
+        """
+        if not (0 <= integer <= 255):
+            raise RuntimeError("Integer must be in range of 0 to 255 inclusive")
+        self.set_digital(self._byte_to_boolean_list(integer, reverse=lsb))
+
+    def set_digital_modes(self, modes: list[bool]):
+        """
+        Set digital pin mode (D0-D7)
+        :param modes: boolean list (LSB first, True = Output, False = Input)
+        """
+        if len(modes) != 8:
+            raise RuntimeError("Need to specify list of 8 boolean values")
+        self.write_value("digital_dir", [self._boolean_list_to_byte(modes)])
+
+    def set_digital_modes_byte(self, mode_int: int, lsb=True):
+        """
+        Set digital output pins mode (D0-D7) from byte
+        :param mode mode_int: Byte (0-255 inclusive)
+        :param lsb: Whether to output in LSB order (0x01 = pin B0)
+        """
+        if not (0 <= mode_int <= 255):
+            raise RuntimeError("Integer must be in range of 0 to 255 inclusive")
+        self.set_digital_modes(self._byte_to_boolean_list(mode_int, reverse=lsb))
+
+    def set_strobe_output(self, mode_int: int):
+        """
+        Strobe output ??
+        It's unclear what the Strobe output does. It appears to just output like the output pins do.
+        S0 = 1, S1 = 2 S2 = 4 S3 = 8 RD = 16 WR = 32 (64 and 128 have seemingly no meaning??)
+        :param mode_int: mode_int: Byte (0-63 inclusive)
+        """
+        if not (0 <= mode_int <= 255):
+            raise RuntimeError("Integer must be in range of 0 to 255 inclusive")
+        self.write_value("strobe_out", [mode_int])
+
+    def set_led_output(self, red=False, blue=False):
+        """
+        Set LED output. True = On, False = Off
+        :param red: Boolean for Red LED
+        :param blue: Boolean for Blue LED
+        """
+        output_byte = (red * 0x01) + (blue * 0x02)
+        self.write_value("led_out", [output_byte])
+
+    def analog_out(self, pin: int, mode: AnalogOutputMode | int, freq: int, voltage_bits: int):
```

Comment:
The `AnalogOutputMode | int` syntax is supported since python 3.10, nxt-python still supports python 3.7.

I would drop the int and only accept an AnalogOutputMode.

Reply:
Fixed in most recent commit.